engine/src/main/java/org/camunda/bpm/engine/impl/cmd/SetExecutionVariablesCmd.java

```diff
@@ -52,9 +62,13 @@ protected ExecutionEntity getEntity() {
       .getExecutionManager()
       .findExecutionById(entityId);
 
-    ensureNotNull("execution " + entityId + " doesn't exist", "execution", execution);
+    if (failIfNotExists) {
+      ensureNotNull("execution " + entityId + " doesn't exist", "execution", execution);
+    }
 
-    checkSetExecutionVariables(execution);
+    if(execution != null) {
+      checkSetExecutionVariables(execution);
```

Comment:
🔧 The `checkSetExecutionVariables` can be moved into the first conditional block after the `ensureNotNull`.

If `failsIfNoExists` is `true` and execution is not null using the `ensureNotNull` guard clause, then the flow can proceed to `checkSetExecutionVariables`.

The second check for execution not being null is not needed.

Reply:
I don't think this keep the desired behavior. If `failIfNotExists = false`, but the object process exists, we want to do `checkSetExecutionVariables`. Your proposed solution wouldn't execute it.